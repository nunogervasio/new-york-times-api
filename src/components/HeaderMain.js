import React from "react";
import { Link } from "react-router-dom";

const HeaderMain = ({ title }) => {
  return (
    <header className="header">
      <Link to="/">
        <h1 className="name">{title}</h1>
      </Link>
    </header>
  );
};

export default HeaderMain;
