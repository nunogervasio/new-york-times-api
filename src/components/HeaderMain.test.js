import React from "react";
import { shallow } from "enzyme";
import HeaderMain from "./HeaderMain";

describe("HeaderMain", () => {
  const headerMain = shallow(<HeaderMain />);

  it("renders properly", () => {
    expect(headerMain).toMatchSnapshot();
  });
});
