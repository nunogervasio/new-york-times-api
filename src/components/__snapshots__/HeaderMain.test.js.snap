// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMain renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderMain />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        to="/"
      >
        <h1
          className="name"
        />
      </Link>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h1
          className="name"
        />,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          to="/"
        >
          <h1
            className="name"
          />
        </Link>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="name"
          />,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "name",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
